docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+a <code>ShoppingList</code> object to keep track of the <code>Item</code> objects the user has added to his list and
+a <code>Budget</code> object to store the user's budget.
+
+<code>Shoco</code> also has a <code>Storage</code> object for saving and loading data from memory - this data includes
+the latest saved <code>ShoppingList</code> and <code>Budget</code>.
+
+There is a dependency from <code>Shoco</code> to <code>Parser</code> as it only creates an instance of the <code>Parser</code>
+every time user input is received by the <code>Ui</code> and does not keep track of the <code>Parser</code> which is deleted
+after it is done parsing the current user input. The <code>Parser</code> determines what command is being invoked by the
+user before creating a new <code>Command</code> object. It then returns the reference to the new <code>Command</code> object 
+to <code>Shoco</code>. 
+
+At any point in time, <code>Shoco</code> only stores up to one <code>Command</code> and no more. This
+<code>Command</code> has to be executed before <code>Shoco</code> can receive more user input.
+<!-- @@author -->
+&nbsp;
+
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+&nbsp;
+
+## 3. Implementation
+This section will describe how the main features of the application are implemented.
+
+### 3.1 Add feature
+#### 3.1.1 Current implementation
+ 
+ The add feature is implemented using an <code>AddCommand</code> class. This class extends from the main
+ <code>Command</code> class. The user input **must contain at least a description** out of these parameters: 
+ *description*, *price*, *quantity*. User can choose not to input price or quantity as the price will set to 
+ default which is 0.0 if the user did not input any value for price. On the other hand, quantity will set to 
+ default which is 1 if the user did not input any value for quantity. 
+ 
+ Process of object creation:
+ 1. <code>Duke</code> class receives user input from the <code>Ui</code> class. 
+ 2. A <code>Parser</code> object is created to call its <code>parseCommand</code> method.
+     * The <code>Parser</code> object instantiates an <code>AddCommand</code> object based on the user input.
+ 3. The <code>Duke</code> class calls the <code>execute</code> method of the <code>AddCommand</code> object.
+ 4. In the <code>execute</code> function, the <code>item</code> to be add is called from the <code>ShoppingList</code> 
+    object, using items.add().
+ 5. In the SD, the AddCommand will add <code>item</code> if the description is provided and one / both price and 
+    quantity is provided. 
+ 6. The <code>item</code> object with its' values is stored into the <code>ShoppingList</code> object.
+ 
+ The following sequence diagram below shows how the add feature works. The details of the adding item's values
+ are shown in a separate sequence diagram below:
+ 
+ ![alt text](images/AddFeature.png)
```

Comment:
Is it supposed to add an object creation mark here? 
<img width="585" alt="image" src="https://user-images.githubusercontent.com/38337361/77997982-eee20300-7362-11ea-83f8-bf69e542c632.png">